### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by alexdavid**
General cleanup and refactoring.

@kevgo @charlierudolph what do you think?

### CODE REVIEW

File: test/spec/exprestive_spec.coffee [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Exprestive = require '../../src/exprestive'
 
 
-describe 'Exprestive configuration', ->
+xdescribe 'Exprestive configuration', ->
```
Comment (alexdavid) [inline test/spec/exprestive_spec.coffee]: These changes cause unit tests to fail.  If we like these changes I'll go ahead and rewrite them

File: src/controller_initializer.coffee [hunk 2]
```diff
@@ -4,10 +4,12 @@ fs = require 'fs'
 path = require 'path'
 
 
-class ControllerStore
+# Initializes controllers in @controllersDir and supports calling saved controller actions
+class ControllerInitializer
 
   constructor: (@controllersDir, @dependencies) ->
     @controllers = {}
+    @initialize()
```
Comment (kevgo) [inline src/controller_initializer.coffee]: A better name for this method seems `loadControllers` or `initializeControllers`

File: src/exprestive.coffee [hunk 3]
```diff
@@ -25,111 +23,43 @@ class Exprestive
...
-    # Object to store controllers
-    @controllers = new ControllerStore @controllersDirPath, @options.dependencies
-
     # Save reverse routes
     @reverseRoutes = @options.routes
-    @middlewareRouter.use @setReverseRoutesOnReqLocals
+
+    # Initialize controllers and routes
+    @controllers = new ControllerInitializer controllersDirPath, @options.dependencies
+    {routes} = new RoutesInitializer routesFilePath, @reverseRoutes
```
Comment (kevgo) [inline src/exprestive.coffee]: It's a little unusual that an object instance is a hash here, or you use only a part of the instance. Can you write this like so:

``` coffeescript
routes = new RoutesInitializer(routesFilePath, @reverseRoutes).routes()
```

File: src/routes_initializer.coffee [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    (url, {to, as}) =>
+      [controllerName, actionName] = to.split '#'
+      @routes.push {httpMethod, url, controllerName, actionName}
+      @registerReverseRoute {routeName: as, httpMethod, url} if as?
+
+
+  # Returns the object of directives passed to the routes file function
+  getRouteDirectives: ->
+    directives = resources: @resourcesDirective
+    for httpMethod in ['GET', 'POST', 'PUT', 'DELETE']
```
Comment (kevgo) [inline src/routes_initializer.coffee]: Extract this magic array into a constant

File: src/routes_initializer.coffee [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  # Save a function to @reverseRoutes to get a url back from a route name
+  registerReverseRoute: ({routeName, httpMethod, url}) ->
+    formatter = new URLFormatter url
+    @reverseRoutes[routeName] = ->
+      formatter.getRoute httpMethod, arguments
+
+
+  # Full list of resource action names. It is important that 'new' precede 'show'
+  resourceActions:
+    ['index', 'new', 'show', 'edit', 'update', 'create', 'destroy']
```
Comment (kevgo) [inline src/routes_initializer.coffee]: Extract this into a constant?

File: src/exprestive.coffee (hunk 3 above)
Comment (alexdavid) [inline src/exprestive.coffee]: What do you think about this

``` coffeescript
routes = new RoutesInitializer routesFilePath, @reverseRoutes
@addRoute route for route in routes.asArray()
```

Comment (kevgo) [inline src/exprestive.coffee]: Sounds good! Maybe name it `toArray()`.

**Pull request closed without merging**